src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Debug/ProjectLaunchTargetsProvider.cs

```diff
@@ -70,14 +68,21 @@ public ProjectLaunchTargetsProvider(
         _threadingService = threadingService;
         _debugger = debugger;
         _remoteDebuggerAuthenticationService = remoteDebuggerAuthenticationService;
-        _hotReloadSessionManager = hotReloadSessionManager;
         _debuggerSettings = debuggerSettings;
         _projectHotReloadLaunchProvider = projectHotReloadLaunchProvider;
     }
 
-    private Task<ConfiguredProject?> GetConfiguredProjectForDebugAsync()
+    private async ValueTask<ConfiguredProject> GetConfiguredProjectForDebugAsync()
     {
-        return _activeDebugFramework.GetConfiguredProjectForActiveFrameworkAsync();
+        var project = await _activeDebugFramework.GetConfiguredProjectForActiveFrameworkAsync();
+        Assumes.NotNull(project);
+        return project;
+    }
+
+    private async ValueTask<IProjectHotReloadSessionManager> GetActiveDebugProjectHotReloadSessionManagerAsync()
```

Comment:
MEF import IActiveConfiguredValue<IProjectHotReloadSessionManager>?

Reply:
As discussed offline, using IActiveConfiguredValue is incorrect since it uses the current scope's ConfiguredProject and we need to use the project returned by `GetConfiguredProjectForDebugAsync()` here.